accepted/00101-native-appstreams.md

```diff
@@ -0,0 +1,75 @@
+
+3. **Package cache**: Incorporates data from repo-sync and package profile update into existing queries, ensuring proper filtering of non-enabled modular packages. The existing tables, queries and stored procedures will be updated with additional joins and proper indexing to ensure minimum performance impact.
+
+A prototype showcasing the core functionality can be found [here](https://github.com/cbbayburt/uyuni/tree/native_appstreams_poc).
+
+### ER diagram
+
+The following figure shows the proposed ER diagram for the core design.
+
+![ER diagram](images/00101-native-appstreams-er-diagram.png)
+
+## New UI operations
+
+The proposed UI includes a new page under the `System > Software` tab, visible only for clients with modular channels assigned. It displays available modules and their streams, allowing users to select and apply module changes to clients.
+
+The main workflow is as follows:
+
+1. Select an assigned modular channel.
+2. Select enabled modules and their streams.
+3. *Optional:* Run `mgr-libmod` for dependency resolution with real-time feedback.
+4. Schedule an Uyuni action to apply the selection on the client.
+5. The action calls `dnf` with the selection of module streams on the client via Salt.
+6. On job return, update the selection of modules in the database.
+
+To allow working on multiple clients at once, SSM will be used. In the SSM, the clients in the set will be first grouped by the modular channels they are assigned to. Then the user will select a group and makes a selection of modules that applies to the whole group.
+
+## Content lifecycle management
+
+The current CLM approach will be replaced with regular *ALLOW/DENY* logic in AppStream filters. During build, module filters will be translated to package filters that apply to all packages of the module. A module will only be included in the target repository if all its packages are visible.
+
+During repository metadata generation, `libmodulemd` will parse and index the modular metadata from the source channels and generate the modular metadata for the target repository following the rules enforced by the attached filters.
+
+If all the modules are denied as a result of the enabled filters, no module metadata will be generated in the target repository.
+
+## Testing
+
+A "dummy" modular repository will be built in OBS for comprehensive testing in the CI environment.
+
+Below is an example of a minimal test repository:
+
+![Test repository](images/00101-native-appstreams-test-repo.png)
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+- Migration plan for the typical user
```

Comment:
While I know this is in the unresolved questions section, are we roughly sure we can migrate everything we currently support in a reasonable way?

Reply:
The new approach is fundamentally different so we won't be able to offer an automated migration at all. So what I mean here is the migration of the customer's workflow.

We just need to make sure that the customers have a smooth way of switching without any disruptions with proper documentation. But in the end, the change is really just getting rid of many extra steps to achieve the same goal, so I'm sure the customers would be happy to go through the labor of migration.

Reply:
While I acknowledge that customers may welcome the transition to the new approach, we cannot compel them to do so by breaking stuff; it should be done at their convenience. In other words, we must ensure that their existing clm project setups remain functional. 

Moreover, if it helps, we could consider applying this change solely to new CLM projects, potentially easing concerns regarding existing setups.

Reply:
Sure, it's a bit tricky because we need to replace the old AppStream filters with the new functionality, but at least we can implement it so their existing projects won't be affected until they want to rebuild them.

Reply:
Would it be possible to implement it in a way that allows uses to continue using the flatten method and this one if they want?
Some customers may be familiar with the old one and may want to continue using it. Abid do you think such a case could exist?

Reply:
Flattening is only possible with the old-style AppStream filters. If we include both old and the new filters, it would be too confusing for users, and also they will keep using the old way for a longer time. That means more L3s and harder maintenance for us so I wouldn't recommend it.

Reply:
Got it, and sounds good to me. :+1: